### PR TITLE
Bazel --config=unit for kubernetes master branch test jobs

### PR DESCRIPTION
**Pull request by BenTheElder**
/assign @ixdy 
cc @mikedanese 

/area bazel
/area jobs

### CODE REVIEW

**Comment by BenTheElder** [comment]
not done, need to tweak the scenario further... ><

**Comment by BenTheElder** [comment]
xref: https://github.com/kubernetes/kubernetes/pull/59283

**Comment by BenTheElder** [comment]
should work, not a huge fan of more flags though :|

File: scenarios/kubernetes_bazel.py [hunk 1]
```diff
@@ -190,6 +190,8 @@ def main(args):
         if test_pkgs or manual_test_targets or affected:
             tests = bazel.query('test', test_pkgs, affected) + manual_test_targets
             if tests:
+                if args.test_config:
+                    tests = ['--config='+args.test_config] + tests
```
Comment (ixdy) [inline scenarios/kubernetes_bazel.py]: is there a strong advantage to doing this rather than just adding `--config=unit` to the test_args option?

Comment (BenTheElder) [reply scenarios/kubernetes_bazel.py]: It seemed maybe more obvious to have a --config flag but in retrospect this is specific to the test portion anyhow so I will do this.

Comment (BenTheElder) [reply scenarios/kubernetes_bazel.py]: updated

File: prow/config.yaml [hunk 2]
```diff
@@ -1239,6 +1240,7 @@ presubmits:
         - "--upload=gs://kubernetes-jenkins/pr-logs"
         - "--" # end bootstrap args, scenario args below
         - "--batch" # mitigation for https://github.com/bazelbuild/bazel/issues/4311
+        - "--test-args=--config=unit"
```
Comment (ixdy) [inline prow/config.yaml]: looks like this needs to be moved down a few lines 🙃

Comment (BenTheElder) [reply prow/config.yaml]: ahh copy paste is evil!
fixed

**Comment by BenTheElder** [comment]
should pass this time
/shrug

**Comment by ixdy** [comment]
/lgtm
/hold

maybe squash?

**Comment by BenTheElder** [comment]
:shrug: makes sense for this PR but generally I have been treating `hack/update-config.sh` like `hack/update-deps.sh`: it has a higher likelihood of conflicts and is auto generated / doesn't need human/manual review unless the generator itself has been changed.

**Pull request merged**